libyul/backends/wasm/TextTransform.cpp

```diff
@@ -65,7 +66,12 @@ string TextTransform::run(wasm::Module const& _module)
 
 string TextTransform::operator()(wasm::Literal const& _literal)
 {
-	return "(i64.const " + to_string(_literal.value) + ")";
+	if (holds_alternative<uint32_t>(_literal.value))
+		return "(i32.const " + to_string(get<uint32_t>(_literal.value)) + ")";
```

Comment:
You're using `get` here, but not in BinaryTransform.

Reply:
I am using `get` in both. In `BinaryTransform` I just need an extra `static_cast` to `int32_t` when calling `lebEncodeSigned(int64_t)` to prevent negative literal values (stored as `uint32_t`) from being encoded as large positive values.